### PR TITLE
add clickBuffer option, prevent default options from being modified, …

### LINKED ISSUE #322: Increase grabsize
.5 is too small:
https://github.com/mapbox/mapbox-gl-draw/blob/e818634859aab8e324f2d85ff3e65a08aac28192/src/lib/features_at.js#L24

I want a value of closer to 2 for the project I am working on now. Should I make this value configurable of just set it to 2? cc: @mcwhittemore

### PR DESCRIPTION
**Pull request by samanpwbb**
…closes #322 
@mcwhittemore for review

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ option | values | function
 --- | --- | ---
 drawing | boolean | The ability to draw and delete features - default: `true`
 keybindings | boolean | Keyboard shortcuts for drawing - default: `true`
+clickBuffer | number | when clicking on a feature, include clicks beyond feature in all directions by this value - default: `2`
```
Comment (mcwhittemore) [inline API.md]: `beyond feature` should be `beyond mouse click`

File: test/options.test.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ test('Options test', t => {
       defaultMode: 'simple_select',
       position: 'top-left',
       keybindings: true,
+      clickBuffer: 2,
```
Comment (mcwhittemore) [inline test/options.test.js]: why is this set to 2?

Comment (samanpwbb) [inline test/options.test.js]: 2 is the default value.

Comment (mcwhittemore) [inline test/options.test.js]: oh, nvm, the cut off confused me

**Comment by mcwhittemore** [comment]
Looks good @samanpwbb

**Pull request merged**